src/CSnakes.SourceGeneration/ResultConversionCodeGenerator.cs

```diff
@@ -3,197 +3,129 @@
+        ParseName("global::CSnakes.Runtime.Python.Internals.PyObjectImporters");
 
-        public TypeSyntax TypeSyntax { get; } = syntax;
-
-        public IEnumerable<StatementSyntax> GenerateCode(ResultConversionCodeGeneratorContext context,
-                                                         string inputName, string outputName) =>
-            [ParseStatement($"var {outputName} = {InternalServices}.{method}({inputName});")];
-    }
-
-    private sealed class RuntimeConversionGenerator(TypeSyntax syntax) :
-        IResultConversionCodeGenerator
+    public static IResultConversionCodeGenerator Create(PythonTypeSpec pythonTypeSpec)
     {
-        public TypeSyntax TypeSyntax => syntax;
-
-        public IEnumerable<StatementSyntax> GenerateCode(ResultConversionCodeGeneratorContext context,
-                                                         string inputName, string outputName) =>
-        [
-            LocalDeclarationStatement(
-                VariableDeclaration(TypeSyntax,
-                    SingletonSeparatedList(
-                        VariableDeclarator(Identifier(outputName))
-                            .WithInitializer(
-                                EqualsValueClause(
-                                    InvocationExpression(
-                                            MemberAccessExpression(
-                                                SyntaxKind.SimpleMemberAccessExpression,
-                                                IdentifierName(inputName),
-                                                GenericName(Identifier("As"), TypeArgumentList(SingletonSeparatedList(TypeSyntax))))))))))
-        ];
+        switch (pythonTypeSpec)
+        {
+            case { Name: "int" }: return Long;
+            case { Name: "str" }: return String;
+            case { Name: "float" }: return Double;
+            case { Name: "bool" }: return Boolean;
+            case { Name: "bytes" }: return ByteArray;
+            case { Name: "Buffer" or "typing.Buffer" or "collections.abc.Buffer" }: return Buffer;
+
+            case { Name: "list" or "typing.List" or "List", Arguments: [var t] }:
+            {
+                return ListConversionGenerator(t);
+            }
+            case { Name: "typing.Sequence" or "Sequence", Arguments: [var t] }:
+            {
+                return ListConversionGenerator(t, "Sequence");
+            }
+            case { Name: "tuple" or "typing.Tuple" or "Tuple", Arguments: { Length: >= 1 and <= 12 } ts }:
+            {
+                var generators = ImmutableArray.CreateRange(from t in ts select Create(t));
+                return new ConversionGenerator(TupleType(SeparatedList(from item in generators select TupleElement(item.TypeSyntax))),
+                                               TypeReflection.CreateGenericType("Tuple", [.. from item in generators select item.TypeSyntax, .. from item in generators select item.ImporterTypeSyntax]));
```

Comment:
I defer to @tonybaloney for the final say, but this mixture of linq and pattern matching is very complicated to tease out exactly the intent or if there is a logical bug.

Reply:
I guess my biggest gripe is this is not debuggable. So if it is correct, w00t. If not, good luck stepping through it.